<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>884</width>
    <height>491</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>884</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Random Walk Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>691</width>
      <height>431</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="central_layout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="information_widget">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="conf_tab">
        <attribute name="title">
         <string>Configuration</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>591</width>
           <height>181</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="conf_layout">
          <item row="0" column="0">
           <widget class="QGroupBox" name="starting_number_box">
            <property name="title">
             <string>Starting Number</string>
            </property>
            <widget class="QLabel" name="starting_number_label">
             <property name="geometry">
              <rect>
               <x>40</x>
               <y>30</y>
               <width>171</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QGroupBox" name="random_walk_box">
            <property name="title">
             <string>Random Walk Number</string>
            </property>
            <widget class="QLabel" name="random_walk_label">
             <property name="geometry">
              <rect>
               <x>50</x>
               <y>30</y>
               <width>171</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QGroupBox" name="nagetive_range_box">
            <property name="title">
             <string>Nagetive Range</string>
            </property>
            <widget class="QLabel" name="nagetive_range_label">
             <property name="geometry">
              <rect>
               <x>40</x>
               <y>30</y>
               <width>171</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QGroupBox" name="positive_range_box">
            <property name="title">
             <string>Positive Range</string>
            </property>
            <widget class="QLabel" name="positive_range_label">
             <property name="geometry">
              <rect>
               <x>50</x>
               <y>30</y>
               <width>171</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QGroupBox" name="random_walk_distance_box">
            <property name="title">
             <string>Random Walk Distance</string>
            </property>
            <widget class="QLabel" name="random_walk_distance_label">
             <property name="geometry">
              <rect>
               <x>40</x>
               <y>26</y>
               <width>171</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="graph_tab">
        <attribute name="title">
         <string>Graph</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="result_tab">
        <attribute name="title">
         <string>Result</string>
        </attribute>
        <widget class="QWidget" name="verticalLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>661</width>
           <height>381</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="result_layout">
          <item>
           <widget class="RandomWalkTable" name="result_table"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="statistics_tab">
        <attribute name="title">
         <string>Statistics</string>
        </attribute>
        <widget class="QWidget" name="gridLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>70</y>
           <width>581</width>
           <height>241</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="statistics_layout">
          <item row="3" column="0">
           <widget class="QLabel" name="tenth_percentile_label">
            <property name="text">
             <string>10'th percentile</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="mean_label">
            <property name="text">
             <string>Mean value</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="standard_deviation_label">
            <property name="text">
             <string>Standard Deviation</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="maximum_label">
            <property name="text">
             <string>Minimum Value</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="minimum_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="tenth_percentile_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="mean_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="standard_deviation_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="minimum_label">
            <property name="text">
             <string>Macimum Value</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="maximum_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="ninetieth_percentile_label">
            <property name="text">
             <string>90'th percentile</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="ninetieth_percentile_value">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QComboBox" name="dimension_statistics_box">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>First D</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Second D</string>
          </property>
         </item>
        </widget>
        <widget class="QComboBox" name="step_statistics_box">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>20</y>
           <width>161</width>
           <height>23</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>if the number is not int. Then it will be rounded.</string>
         </property>
         <item>
          <property name="text">
           <string>10% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>20% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>30% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>40% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>50% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>60% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>70% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>80% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>90% of random step</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>100% of random step</string>
          </property>
         </item>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="preference_widget">
   <property name="minimumSize">
    <size>
     <width>168</width>
     <height>449</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>168</width>
     <height>449</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable|QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <property name="windowTitle">
    <string>Preferences</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="preference_widget_contents">
    <widget class="QGroupBox" name="dimension_box">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>120</width>
       <height>71</height>
      </rect>
     </property>
     <property name="title">
      <string>Dimension</string>
     </property>
     <widget class="QComboBox" name="dimension_value">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>One D</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Two D</string>
       </property>
      </item>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="action_new"/>
   <addaction name="action_edit"/>
   <addaction name="separator"/>
   <addaction name="action_start"/>
   <addaction name="action_stop"/>
   <addaction name="action_clear"/>
  </widget>
  <action name="action_new">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/new_file.png</normaloff>
     <normalon>:/res/save.png</normalon>:/res/new_file.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="action_edit">
   <property name="icon">
    <iconset>
     <normalon>:/res/save.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
  </action>
  <action name="action_start">
   <property name="icon">
    <iconset>
     <normalon>:/res/start.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </action>
  <action name="action_clear">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/res/reset.png</normaloff>:/res/reset.png</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="action_stop">
   <property name="icon">
    <iconset>
     <normalon>:/res/stop.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RandomWalkTable</class>
   <extends>QTableWidget</extends>
   <header>random_walk_table</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>dimension_value</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>change_dimension(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>819</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>709</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dimension_statistics_box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refresh_statistics_tab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>187</x>
     <y>44</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>step_statistics_box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refresh_statistics_tab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>418</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refresh_statistics_tab(int)</slot>
  <slot>change_dimension(int)</slot>
 </slots>
</ui>
